pkg/resource/deploy/plan_executor.go

```diff
@@ -269,7 +269,11 @@ func (pe *planExecutor) Execute(callerCtx context.Context, opts Options, preview
 	if res == nil && !pe.stepExec.Errored() {
 		res := pe.stepGen.AnalyzeResources()
 		if res != nil {
-			return res
+			if resErr := res.Error(); resErr != nil {
+				logging.V(4).Infof("planExecutor.Execute(...): error analyzing resources: %v", resErr)
+				pe.reportError("", resErr)
+			}
+			return result.Bail()
```

Comment:
This is along the lines of how errors are reported when individual resources are analyzed, which bubble up to:

https://github.com/pulumi/pulumi/blob/215c3953a7dd826a6f7a14ff30844c40138833ca/pkg/resource/deploy/plan_executor.go#L236-L241